haystack/nodes/prompt/invocation_layer/sagemaker.py

```diff
@@ -0,0 +1,221 @@
+        return generated_texts
+
+    def _post(self, prompt: str, params: Optional[Dict[str, Any]] = None) -> List[str]:
+        """
+        Post data to the SageMaker inference model. It takes in a prompt and returns a list of responses using model invocation.
+        :param prompt: The prompt text to be sent to the model.
+        :param params: The parameters to be sent to the Hugging Face model (see https://huggingface.co/docs/api-inference/detailed_parameters#text-generation-task)
+        :return: The generated responses as a list of strings.
+        """
+
+        try:
+            body = {"inputs": prompt, "parameters": params}
+            response = self.client.invoke_endpoint(
+                EndpointName=self.model_name_or_path,
+                Body=json.dumps(body),
+                ContentType="application/json",
+                Accept="application/json",
+            )
+            response_json = response.get("Body").read().decode("utf-8")
+            output = json.loads(response_json)
+            generated_texts = [o["generated_text"] for o in output if "generated_text" in o]
+            return generated_texts
+        except requests.HTTPError as err:
+            res = err.response
+            if res.status_code == 429:
+                raise SageMakerModelNotReadyError(f"Model not ready: {res.text}")
+            raise SageMakerInferenceError(
+                f"SageMaker Inference returned an error.\nStatus code: {res.status_code}\nResponse body: {res.text}",
+                status_code=res.status_code,
+            )
+
+    def _ensure_token_limit(self, prompt: Union[str, List[Dict[str, str]]]) -> Union[str, List[Dict[str, str]]]:
+        # the prompt for this model will be of the type str
+        resize_info = self.prompt_handler(prompt)  # type: ignore
+        if resize_info["prompt_length"] != resize_info["new_prompt_length"]:
+            logger.warning(
+                "The prompt has been truncated from %s tokens to %s tokens so that the prompt length and "
+                "answer length (%s tokens) fit within the max token limit (%s tokens). "
+                "Shorten the prompt to prevent it from being cut off.",
+                resize_info["prompt_length"],
+                max(0, resize_info["model_max_length"] - resize_info["max_length"]),  # type: ignore
+                resize_info["max_length"],
+                resize_info["model_max_length"],
+            )
+        return str(resize_info["resized_prompt"])
+
+    @classmethod
+    def supports(cls, model_name_or_path: str, **kwargs) -> bool:
+        """
+        Checks whether a model_name_or_path passed down (e.g. via PromptNode) is supported by this class.
+
+        :param model_name_or_path: The model_name_or_path to check.
+        """
+        client = None
+        try:
+            session = boto3.Session(
+                aws_access_key_id=kwargs.get("aws_access_key_id"),
```

Comment:
In the _ _ init _ _ method, these five parameters are `Optional[str]`, so what happens if they are all None? Will `supports` return `False` for every model then?

Reply:
No, it can still work if all of the args are None. boto3 falls back to the default profile and credentials of the machine if none of the args is provided here. This means, whatever you have defined in ~/.aws/credentials and ~/.aws/config